browser/brave_browser_main_extra_parts.cc

```diff
@@ -15,3 +16,7 @@ BraveBrowserMainExtraParts::~BraveBrowserMainExtraParts() {
 void BraveBrowserMainExtraParts::PreCreateThreads() {
   brave::InitializeResourceBundle();
 }
+
+void BraveBrowserMainExtraParts::PreMainMessageLoopRun() {
```

Comment:
this is fine because the profile is initialized in PreMainMessageLoopRun and this will run after the chrome extra parts, but I think we want to guard this with local_state pref that is set after import

Reply:
Do you think guarding with a local_state pref would be preferable to conditioning auto-import on first run, e.g. via [first_run::IsChromeFirstRun](https://cs.chromium.org/chromium/src/chrome/browser/chrome_browser_main.cc?l=1722-1723&rcl=9a0d2a34e84c0ad0e6f982bcf83fa7179e56567f)?

Reply:
> instead of using a command line flag

We need a command line flag to distinguish "Brave-Core is running for the first time after being installed by the Muon upgrader" and "Brave-Core is running for the first time after being downloaded and installed de novo". See https://github.com/brave/brave-browser/issues/1545#issuecomment-432715825 for discussion.

Reply:
shouldn't we just upgrade by default if we detect a previous browser-laptop install? Then no command line flag is necessary and a pref would work fine

Reply:
> shouldn't we just upgrade by default if we detect a previous browser-laptop install?

We discussed this in a meeting last week and consensus was that auto-import should only happen in the case where the user is running browser-laptop and performs the update that migrates them to brave-core. We need a way to distinguish this from the case where the user downloads and installs brave-core separately, but may also have an existing browser-laptop installation (which could be their current/default browser, or defunct and unused and therefore probably not desirable to auto-import from). We agreed that passing a command line argument to the brave-core binary from the auto-update was the best way to distinguish these cases.

cc @bsclifton @davidtemkin

Reply:
@bridiver What do you think of the local state pref I added in 51c68d4? With this addition,

1. auto-import from Muon only occurs if triggered by passing `--upgrade-from-muon` command line flag
2. auto-import from Muon status is tracked in a local_state pref, so auto-import can occur at most once
3. the local_state pref may be useful to other components (e.g. if Brave Rewards wants to convert pins -> donations, it can check the pref to see if there are any imported pins that need conversion)